### PR TITLE
Extracted the debug info generation for functions into a cmake option

### PR DESCRIPTION
**Pull request by myrrc**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Fixed the backward-incompatible change by providing the options to build without debug info for functions.

### CODE REVIEW

**Review by alexey-milovidov — CHANGES_REQUESTED** [review]
Inconsistent with https://github.com/ClickHouse/ClickHouse/pull/14549/files

File: src/Functions/CMakeLists.txt [hunk 1]
```diff
@@ -53,8 +53,28 @@ endif()
...
+    set (STRIP_DSF_DEFAULT OFF)
+endif()
+
+
+option(STRIP_DEBUG_SYMBOLS_FUNCTIONS
+    "Do not generate debugger info for ClickHouse functions.
+    Provides faster linking and lower binary size.
+    Tradeoff is the inability to debug some source files with e.g. gdb
+    (empty stack frames and no local variables)."
+    STRIP_DSF_DEFAULT)
```
Comment (azat) [inline src/Functions/CMakeLists.txt:70]: Should this be `${STRIP_DSF_DEFAULT}` ?

Comment (myrrc) [reply src/Functions/CMakeLists.txt]: Thanks!

**Comment by alexey-milovidov** [comment]
> test_adaptive_granularity/test.py::test_version_update_two_nodes

- flaky test @alesapin

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**